shared/constants/types/fs.js

```diff
@@ -2,21 +2,40 @@
 import * as I from 'immutable'
 
 export opaque type Path = ?string
-export type FolderPathType = 'folder'
-export type FilePathType = 'file' | 'symlink' | 'exec'
-export opaque type PathType = FolderPathType | FilePathType
```

Comment:
I think the PathType doesn't have that much value in being an opaque type. I think it confuses flow w/ the disjoint unions and its not something i think you'll get much protection from. In my mind the opaque type lets you enforce if you have multiple id type things from accidentally getting confused because they're compatible (all strings or numbers) but this type file isn't going to be an input to anything really